File: packages/ai-config/src/schema/model.schema.ts
// Esquema de validación para Model
import { z } from 'zod';

export const ModelCompatSchema = z.object({
  context: z.array(z.string()),
  reasoning: z.array(z.string()),
  tools: z.array(z.string()),
  transport: z.array(z.string()),
});

export const ModelSchema = z.object({
  id: z.string(),
  name: z.string(),
  providerId: z.string().optional(),
  compat: ModelCompatSchema,
  meta: z.record(z.string(), z.unknown()).optional(),
});

File: packages/ai_config/schema/model_schema.py
from pydantic import BaseModel
from typing import List, Dict, Optional, Any

class ModelCompatSchema(BaseModel):
    context: List[str]
    reasoning: List[str]
    tools: List[str]
    transport: List[str]

class ModelSchema(BaseModel):
    id: str
    name: str
    provider_id: Optional[str] = None
    compat: ModelCompatSchema
    meta: Optional[Dict[str, Any]] = None
